### ISSUE TITLE
Review usage of `aria-live: assertive`, `alert` throughout the code base

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#185371]
@rperez030 has called to my attention an overuse of `alert`, which uses `aria-live: assertive`, throughout the code base. cc @jooyoungseo

Please review each of your usages:
- Test the experience on macOS/ Windows with a screen reader before making any changes.
- Use `status` (which uses `aria-live: polite`) instead of `alert` and verify the content is still announced as expected.
- If there's a compelling reason to use `alert` instead, it is fine to stick with that. Namely, if the announcement is time sensitive, use `alert`.
- Check off your item below.

Guidance from https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions:

Any region which receives updates that are important for the user to receive, but not so rapid as to be annoying, should receive the aria-live="polite". The screen reader will speak changes whenever the user is idle.

aria-live="assertive" should only be used for time-sensitive/critical notifications that absolutely require the user's immediate attention. Generally, a change to an assertive live region will interrupt any announcement a screen reader is currently making. As such, it can be extremely annoying and disruptive and should only be used sparingly.

Example change:
https://github.com/microsoft/vscode/pull/185368/files#diff-1a26fc97bbf0d39b6d25ad9fe509fad4d90373118985b50709f99b6ef5bde8c9L532

- [x] @bpasero https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/browser/parts/notifications/notificationsAlerts.ts#L75
- [x] @joaomoreno https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/base/browser/ui/list/listWidget.ts#L506 https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/parameterHints/browser/parameterHintsWidget.ts#L264
- [x] @alexdima https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/anchorSelect/browser/anchorSelect.ts#L61 https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/toggleTabFocusMode/browser/toggleTabFocusMode.ts#L39 https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/codeEditor/browser/accessibility/accessibility.ts#L122 
- [x] @rebornix https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/find/browser/findWidget.ts#L430
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/notebook/browser/contrib/find/notebookFindWidget.ts#L367
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts#L2286
- [x] @jrieken https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/format/browser/format.ts#L50-L59 https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/gotoSymbol/browser/goToCommands.ts#L128 https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/message/browser/messageController.ts#L57 https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/rename/browser/rename.ts#L253 https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/suggest/browser/suggestController.ts#L547 https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts#L279
- [x] @hediet https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/inlineCompletions/browser/inlineCompletionsController.ts#L189 
- [x] @meganrogge 
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/base/browser/ui/inputbox/inputBox.ts#L499
- [x] @TylerLeonhardt https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/common/editor/textEditorModel.ts#L136
- [x] @roblourens https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/debug/browser/debugService.ts#L910
- [x] @sandy081  many instances throughout each of these files 
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/extensions/browser/extensionsActions.ts#L435
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts#L720
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/extensions/browser/extensionsViews.ts#L1050
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/preferences/browser/keybindingWidgets.ts#L233
- [x] @rzhao271 two instances in this file
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/preferences/browser/settingsTree.ts#L2077
- [x] @andreamah two instances in this file
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/search/browser/searchView.ts#L948
https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/searchEditor/browser/searchEditor.ts#L450
- [x] @connor4312 https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/workbench/contrib/testing/browser/testingOutputPeek.ts#L542

**Comment by jooyoungseo** [comment]
Programming context is unique which entails immediate attention, such as terminal output, suggestions, completions, notification, etc. These directly lead to productivity. Using `assertive` might be appropriate for the most cases. `polite` might introduce unnecessary delay.

**Comment by meganrogge** [comment]
@jooyoungseo Thank you for that input. I think that's why it's important to review and evaluate the merit of each usage. For example, when searching in the terminal, we probably don't need to alert and can instead use status to report `result 1 of 5` etc.

**Comment by rperez030** [comment]
@jooyoungseo  curious to know in which instance you believe the use of polite updates could introduce a delay. In the examples you indicated, (terminal output, suggestion, completion...) the events occur immediately, so a change from assertive to polite should not be perceivable in normal circumstances, except in macOS where we wouldn't hear the audio cue that voiceover generates when assertive is used. This would play a role in situations in which a live update competes with a focus event, for example. When that happens, different screen readers behave differently.

- NVDA: assertive: Announces the live update followed by all other events. Polite: Announces all events, including the live update, in their default priority.
- JAWS: assertive: Places the live update on top of any other aria live update, but below any other event which it considers having higher priority. for example,  focus update would be announced first. Polite: Announces all events in their default priority.
- Voiceover. Assertive: announces the live update and suppresses the rest of the announcements queue. NVDA used to do this until the refactor of the speech subsystem that took place a few years ago. Polite: Announces everything in their default priority, possibly simply in the order they were received.

The above is the result of my own limited testing, except in the case of NVDA where the behavior is properly documented somewhere in Github.

In summary: I believe the use of aria-live="polite" should not introduce more delay than the one we already experience in most situations I can think of.

**Comment by bpasero** [comment]
@meganrogge 

I passed https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/base/browser/ui/inputbox/inputBox.ts#L499 back to you, the origins are in https://github.com/microsoft/vscode/pull/92531, reviewed by Isidor.

As for the usage in notification toasts, I would need your guidance if the usage is OK or not. Often notification toasts require user attention, for example in some flows where a user triggers an action, the next step requires using one of the buttons in a notification. I wonder if the use of `alert` in this case is actually OK. But I am also fine changing it.

**Comment by meganrogge** [comment]
@bpasero I think it makes sense to use `alert` for notifications because they're often time sensitive / important.

**Comment by joaomoreno** [comment]
Same here, @isidorn can you double check the ones in [listWidget.ts](https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/base/browser/ui/list/listWidget.ts#L506) and [parameterHintsWidget.ts](https://github.com/microsoft/vscode/blob/4d42fcf80927152e9a92506e7d5df520f61ed05c/src/vs/editor/contrib/parameterHints/browser/parameterHintsWidget.ts#L264)?

**Comment by meganrogge** [comment]
I can @joaomoreno

**Comment by meganrogge** [comment]
code review would be good to verify here
